portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,18 +32,77 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> comments = new ArrayList<>();
-    comments.add("Hey !!! Just commenting - Prakash");
-    comments.add("Roses are red - Nirmala");
-    comments.add("Violets are blue - Sushant");
-    comments.add("What I am doing, I have no clue- Prashant");  
-    
+
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    int maxCommentLimit = getCommentLimitFromParam(request, "max-comment");
+    List<Entity> comments = results.asList(FetchOptions.Builder.withLimit(maxCommentLimit));
+
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String firstName = getParameter(request, "first-name", "");
+    String lastName = getParameter(request, "last-name", "");
+    String commentText = getParameter(request, "comment", "");
+    Date commentDate = new Date();
+
+    Entity commentEntity = new Entity("Comment");
+
+    commentEntity.setProperty("firstName", firstName);
+    commentEntity.setProperty("lastName", lastName);
+    commentEntity.setProperty("commentText", commentText);
+    commentEntity.setProperty("date", commentDate.toString());
+
+    datastore.put(commentEntity);
+    
+    response.sendRedirect("/index.html#connect");
+  }
+  
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").setKeysOnly();
+
+    PreparedQuery results = datastore.prepare(query);
+    
+    for (Entity entity : results.asIterable()) {
+      datastore.delete(entity.getKey());
+    }
+
+    response.getWriter().println("Successfully deleted all entries");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  private int getCommentLimitFromParam(HttpServletRequest request, String parameter) {
```

Comment:
This method should likely be directly under the doGet() method since that is what calls this method. 

See http://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents

Reply:
Thanks for the reference.
Changes Made.